cts-java/src/test/java/org/ga4gh/cts/api/reads/ReadsSearchIT.java

```diff
@@ -0,0 +1,263 @@
+
+        final List<ReadAlignment> alignments = srResp.getAlignments();
+        alignments.stream().forEach(read -> assertThat(read.getNextMatePosition()).isNotNull());
+        alignments.stream()
+                  .forEach(read -> assertThat(read.getNextMatePosition()
+                                                  .getReferenceName()).isEqualTo(TestData.REFERENCE_NAME));
+        alignments.stream().forEach(read -> assertThat(read.getAlignment()).isNotNull());
+        alignments.stream().forEach(read -> assertThat(read.getAlignment().getCigar()).isNotNull());
+    }
+
+    /**
+     * Fetch every {@link ReadGroupSet} named in {@link TestData#EXPECTED_READGROUP_NAMES} using
+     * <tt>searchReadGroupSets</tt> and verify that it returns the expected {@link ReadGroupSet}.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void searchReadGroupSetsMustReturnReadGroupSetsWithExpectedNames() throws
+            AvroRemoteException {
+
+        for (String expectedReadGroupSetName : TestData.EXPECTED_READGROUPSETS_NAMES) {
+
+            final SearchReadGroupSetsRequest req =
+                    SearchReadGroupSetsRequest.newBuilder()
+                                              .setDatasetId(TestData.getDatasetId())
+                                              .setName(expectedReadGroupSetName)
+                                              .build();
+            final SearchReadGroupSetsResponse resp = client.reads.searchReadGroupSets(req);
+
+            final List<ReadGroupSet> readGroupSets = resp.getReadGroupSets();
+
+            assertThat(readGroupSets).hasSize(1);
+            final ReadGroupSet readGroupSet = readGroupSets.get(0);
+            assertThat(readGroupSet.getName()).isEqualTo(expectedReadGroupSetName);
+            assertThat(readGroupSet.getDatasetId()).isEqualTo(TestData.getDatasetId());
+        }
+    }
+
+    /**
+     * Verify that passing zero read group names in a {@link SearchReadsRequest} fails.
+     *
+     * @throws Exception if there's a problem we didn't catch
+     */
+    @SuppressWarnings("ThrowableResultOfMethodCallIgnored")
+    @Test
+    public void searchReadsWithNoIdsFails() throws Exception {
+
+        // first get a valid reference
+        final String refId = Utils.getValidReferenceId(client);
+
+        final SearchReadsRequest request =
+                SearchReadsRequest.newBuilder()
+                                  .setReadGroupIds(Collections.emptyList())
+                                  .setStart(0L)
+                                  .setEnd(150L)
+                                  .setReferenceId(refId)
+                                  .build();
+
+        final GAWrapperException t = catchGAWrapperException(() -> client.reads.searchReads(request));
+        assertThat(t.getHttpStatusCode()).isEqualTo(HttpURLConnection.HTTP_NOT_IMPLEMENTED);
```

Comment:
I respect that.  I had updated the schemas `documentation` branch to document this behavior, since it was previously utterly undefined and undocumented.  Please feel free to open an issue against the schemas to covert this; the compliance suite can be made to test for any agreed-upon behavior, of course.